src/shogun/classifier/svm/NewtonSVM.h

```diff
@@ -94,28 +94,21 @@ class CNewtonSVM : public CLinearMachine
 		virtual const char* get_name() const { return "NewtonSVM"; }
 
 	protected:
-		/** train SVM classifier
-		 *
-		 * @param data training data (parameter can be avoided if distance or
-		 * kernel-based classifiers are used and distance/kernels are
-		 * initialized with train data)
-		 *
-		 * @return whether training was successful
-		 */
-		virtual bool train_machine(CFeatures* data=NULL);
+		virtual void init_model(CFeatures* data);
+		virtual void iteration();
+		virtual void end_training();
 
 	private:
-		void obj_fun_linear(float64_t* weights, float64_t* out, float64_t* obj,
-				int32_t* sv, int32_t* numsv, float64_t* grad);
+		void obj_fun_linear();
 
-		void line_search_linear(float64_t* weights, float64_t* d,
-				float64_t* out, float64_t* tx);
+		void line_search_linear(float64_t* d);
 
 	protected:
 		/** lambda=1/C */
 		float64_t lambda, C, epsilon;
-		float64_t prec;
-		int32_t x_n, x_d, num_iter;
+		float64_t *weights, *out, *grad;
```

Comment:
which ones should we register here ? :)

Reply:
Only the ones that are important for the model being applied to new data.
Probably only the weights.

You could add this to the serialised model test to find out what makes it work, that’s what the test is for .